static/app/views/issueDetails/groupDetails.tsx

```diff
@@ -253,6 +247,24 @@ function useFetchOnMount({fetchData}: Pick<FetchGroupDetailsState, 'fetchData'>)
   }, []);
 }
 
+function useEventApiQuery(
+  eventID: string,
+  queryKey: [string, {query: {environment?: string[]}}]
+) {
+  const isLatest = eventID === 'latest';
+  const latestEventQuery = useApiQuery<Event>(queryKey, {
+    staleTime: 10000,
+    cacheTime: 30000,
+    enabled: isLatest,
```

Comment:
One additional option you want here that will help the behavior a bit is:

```ts
retry: (_, error) => error.status !== 404
```

React query will automatically retry failed requests, but 404s are expected (like when you change the environment dropdown to one that has no events). So this will show the error state immediately